pkgs/os-specific/windows/cygwin/default.nix

```diff
@@ -10,56 +10,77 @@
   flex,
   perl,
   mingw_w64,
+  symlinkJoin,
 }:
 
-stdenv.mkDerivation {
-  pname = "cygwin";
+let
+  newlib-cygwin = stdenv.mkDerivation {
+    pname = "cygwin";
 
-  inherit (cygwin_headers)
-    version
-    src
-    meta
-    patches
-    ;
+    inherit (cygwin_headers)
+      version
+      src
+      meta
+      patches
+      ;
 
-  preConfigure = ''
-    pushd winsup
-    aclocal --force
-    autoconf -f
-    automake -ac
-    rm -rf autom4te.cache
-    popd
-    patch -p0 -i ${./after-autogen.patch}
-  '';
+    preConfigure = ''
+      pushd winsup
+      aclocal --force
+      autoconf -f
+      automake -ac
+      rm -rf autom4te.cache
+      popd
+      patch -p0 -i ${./after-autogen.patch}
+    '';
 
-  postPatch = ''
-    patchShebangs --build winsup/cygwin/scripts
-  '';
+    postPatch = ''
+      patchShebangs --build winsup/cygwin/scripts
+    '';
 
-  env.CXXFLAGS_FOR_TARGET = "-Wno-error";
+    env.CXXFLAGS_FOR_TARGET = "-Wno-error";
 
-  depsBuildBuild = [ buildPackages.stdenv.cc ];
-  nativeBuildInputs = [
-    autoconf
-    automake
-    bison
-    cocom-tool-set
-    flex
-    perl
-  ];
-  buildInputs = [ mingw_w64 ];
+    depsBuildBuild = [ buildPackages.stdenv.cc ];
+    nativeBuildInputs = [
+      autoconf
+      automake
+      bison
+      cocom-tool-set
+      flex
+      perl
+    ];
+    buildInputs = [ mingw_w64 ];
 
-  hardeningDisable = [ "fortify" ];
-  configurePlatforms = [
-    "build"
-    "target"
-  ];
-  configureFlags = [
-    "--disable-shared"
-    "--disable-doc"
-    "--enable-static"
-    "--disable-dumper"
-    "--with-cross-bootstrap"
-    "ac_cv_prog_CC=gcc"
+    postInstall = ''
+      mv $out/x86_64-pc-cygwin/* $out/
+      rmdir $out/x86_64-pc-cygwin
+    '';
+
+    hardeningDisable = [
+      "fortify"
+      "stackprotector"
+    ];
+    configurePlatforms = [
+      "build"
+      "target"
+    ];
+    configureFlags = [
+      "--disable-shared"
+      "--disable-doc"
+      "--enable-static"
+      "--disable-dumper"
+      "--with-cross-bootstrap"
+      "ac_cv_prog_CC=gcc"
+    ];
+  };
+in
+# TODO: Is there something like nix-support which would achieve this better?
```

Comment:
Could you just link the appropriate paths in postInstall in the main derivation?